### ISSUE TITLE
F5 environment not update after changes k8s ingress resources

### ISSUE THREAD

**Issue by wheestermanstoyota** [F5Networks/k8s-bigip-ctlr#1065]
#### Description
Yesterday we had issues inside the production environment, for some reason the BIGIP controller, deployed in kubernetes, was not updating the LB configuration causing some applications to be inaccessible. We had similar issue in non-production beginning of August. Restarting the controller pod solves all problems.

#### Kubernetes Version
v1.11.10-docker-1 -- Docker-EE 2.1

#### Controller Version
1.10.0

#### Diagnostic Information
-	Docker-EE 2.1 - kubernetes v1.11.10-docker-1
-	BPGPeering has been configured – kubernetes calico 3.8.2
-	BIGIP controller version f5networks/k8s-bigip-ctlr:1.10.0
-	Attached the log files I collected before restarting the pod (bigip.zip)
-	Attached the controller deployment yaml file (bigip.yaml)
-	Attached an Ingress/Service example deployed application (as deployment), all application are deployed like this example (ingress + service yaml)

I attached also log of last evening and today, and I need some explanation about the following log lines:

In red you see some warnings and errors.
In green I see some updates logged (overwriting), why we see these even on moments nothing is changed inside the environment
```
2019/10/14 17:33:59 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-http-cwsweb-ui ServicePort:8080 Namespace:tme-ns-http-prd}
2019/10/14 17:33:59 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-http-cwsweb-ui ServicePort:8080 Namespace:tme-ns-http-prd}
2019/10/14 22:27:14 [ERROR] [2019-10-14 22:27:14,348 __main__ ERROR] Failed to process the config file /tmp/k8s-bigip-ctlr.config636764988/config.json (ValueError: No JSON object could be decoded)
2019/10/14 22:27:14 [ERROR] [2019-10-14 22:27:14,349 __main__ ERROR] Error applying config, will try again in 1 seconds
2019/10/14 23:01:01 [ERROR] [as3_log] REST call error: Post https://10.100.114.168:443/mgmt/shared/appsvcs/declare: x509: cannot validate certificate for 10.100.114.168 because it doesn't contain any IP SANs 
2019/10/14 23:01:03 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-http-cwsweb-ui ServicePort:8080 Namespace:tme-ns-http-prd}
2019/10/14 23:01:04 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-nodejs-rem-backend ServicePort:8080 Namespace:tme-ns-nodejs-prd}
2019/10/14 23:01:04 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-nodejs-pat-backend ServicePort:8080 Namespace:tme-ns-nodejs-prd}
2019/10/14 23:01:05 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-hub-service-app-va ServicePort:8080 Namespace:tme-ns-jetty-prd}
2019/10/14 23:01:06 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-hub-service-app-va ServicePort:8080 Namespace:tme-ns-jetty-prd}
2019/10/14 23:01:07 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-hub-service-app-zb ServicePort:8080 Namespace:tme-ns-jetty-prd}
2019/10/14 23:01:07 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-hub-service-app-zb ServicePort:8080 Namespace:tme-ns-jetty-prd}
2019/10/14 23:01:07 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-hub-service-app-zb ServicePort:8080 Namespace:tme-ns-jetty-prd}
2019/10/14 23:01:07 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-hub-service-app-zb ServicePort:8080 Namespace:tme-ns-jetty-prd}
2019/10/14 23:01:08 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-hub-service-app-zb ServicePort:8080 Namespace:tme-ns-jetty-prd}
2019/10/14 23:01:08 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-hub-service-app-zb ServicePort:8080 Namespace:tme-ns-jetty-prd}
2019/10/14 23:01:08 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-hub-service-app-zb ServicePort:8080 Namespace:tme-ns-jetty-prd}
2019/10/14 23:01:08 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-hub-service-app-zb ServicePort:8080 Namespace:tme-ns-jetty-prd}
2019/10/14 23:01:08 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-hub-service-app-zb ServicePort:8080 Namespace:tme-ns-jetty-prd}
2019/10/14 23:01:09 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-hub-service-app-zb ServicePort:8080 Namespace:tme-ns-jetty-prd}
2019/10/14 23:01:49 [WARNING] Overwriting existing entry for backend {ServiceName:twistlock-console ServicePort:8083 Namespace:twistlock}
2019/10/15 01:13:23 [WARNING] Did not receive config write response in 1s
2019/10/15 02:25:49 [ERROR] [2019-10-15 02:25:49,643 __main__ ERROR] Unexpected error
2019/10/15 02:25:49 [ERROR] [2019-10-15 02:25:49,645 __main__ ERROR] Error applying config, will try again in 1 seconds
```

**Comment by wheestermanstoyota** [comment]
What I also see regularly are warnings about overwriting existing entries, but nothing happened for these, the applications for these are not redeployed inside k8s. For which reason do I see those then?

2019/10/30 12:12:02 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-nodejs-pat-frontend ServicePort:8080 Namespace:tme-ns-nodejs-dev}

2019/10/30 12:12:02 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-nodejs-pat-frontend ServicePort:8080 Namespace:tme-ns-nodejs-dev}

2019/10/30 12:12:02 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-nodejs-pat-frontend ServicePort:8080 Namespace:tme-ns-nodejs-dev}

2019/10/30 12:12:03 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-nodejs-pat-frontend ServicePort:8080 Namespace:tme-ns-nodejs-dev}

Walter

**Comment by wheestermanstoyota** [comment]
Some other sample of the warnings:

2019/10/30 12:22:44 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}
2019/10/30 12:22:44 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}
2019/10/30 12:22:44 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}
2019/10/30 12:22:44 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}
2019/10/30 12:22:45 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}
2019/10/30 12:22:45 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}
2019/10/30 12:22:45 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}
2019/10/30 12:22:45 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}
2019/10/30 12:22:45 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}
2019/10/30 12:22:46 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}
2019/10/30 12:22:46 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}
2019/10/30 12:22:46 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}
2019/10/30 12:22:46 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}
2019/10/30 12:22:46 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}
2019/10/30 12:22:47 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}
2019/10/30 12:22:47 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}
2019/10/30 12:22:47 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}
2019/10/30 12:22:48 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}
2019/10/30 12:22:48 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}
2019/10/30 12:22:48 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}
2019/10/30 12:22:48 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}
2019/10/30 12:22:49 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}
2019/10/30 12:22:49 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}
2019/10/30 12:22:49 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}
2019/10/30 12:22:49 [WARNING] Overwriting existing entry for backend {ServiceName:tme-svc-jetty-vlps-batch-app ServicePort:8080 Namespace:tme-ns-jetty-dev}

Looks to me a strange behavior...

**Comment by veeramalla-f5** [comment]
@wheestermanstoyota .. These warnings can be completely ignored. We will look into a way to get rid of these.

**Comment by wheestermanstoyota** [comment]
Thanks for the confirmation.

So, we still need to find out why we had until now three situations (two in non-production k8s cluster and once in production k8s cluster) that after deployments of applications the F5 configuration didn't got update, causing some pools to be in a red non-healthy state. After restarting the pod of the F5-BIGIP controller solved the issue.

Regards
Walter

**Comment by veeramalla-f5** [comment]
@wheestermanstoyota Did u scale the number of services? Can you provide more information on the number of services your cc/cis controller is monitoring. 

It will be great if you can provide more information on the Error. Any steps to reproduce the Error will help.

**Comment by wheestermans31** [comment]
Currently we have tow k8s clusters, one non-production and one production. 

Inside the non-production cluster we have two controllers running; one looking at the development namespaces and one for the acceptance namespaces. We have these two different F5 LB configurations.

Development controller:

        - --bigip-username=$(BIGIP_USERNAME)
        - --bigip-password=$(BIGIP_PASSWORD)
        - --bigip-url=https://nnn.nnn.nnn.nnn:nnn
        - --bigip-partition=KuberNetes
        - --pool-member-type=cluster
        - --namespace=tme-ns-http-dev
        - --namespace=tme-ns-nodejs-dev
        - --namespace=tme-ns-jetty-dev
        - --namespace=tme-ns-kong-dev
        - --vs-snat-pool-name=KuberNat
        - --manage-configmaps=false
        - --log-level=WARNING
        - --verify-interval=10

Acceptance controller:

        - --bigip-username=$(BIGIP_USERNAME)
        - --bigip-password=$(BIGIP_PASSWORD)
        - --bigip-url=https://nnn.nnn.nnn.nnn:nnn
        - --bigip-partition=KuberNetes
        - --pool-member-type=cluster
        - --namespace=tme-ns-http-uat
        - --namespace=tme-ns-nodejs-uat
        - --namespace=tme-ns-jetty-uat
        - --namespace=tme-ns-http-uat-1
        - --namespace=tme-ns-nodejs-uat-1
        - --namespace=tme-ns-jetty-uat-1
        - --namespace=tme-ns-http-uat-2
        - --namespace=tme-ns-nodejs-uat-2
        - --namespace=tme-ns-jetty-uat-2
        - --namespace=tme-ns-http-uat-8
        - --namespace=tme-ns-nodejs-uat-8
        - --namespace=tme-ns-jetty-uat-8
        - --namespace=tme-ns-http-emr
        - --namespace=tme-ns-nodejs-emr
        - --namespace=tme-ns-jetty-emr
        - --vs-snat-pool-name=KuberNat
        - --verify-interval=10
        - --manage-configmaps=false
        - --log-level=WARNING

Inside the production cluster we have one controller running looking at the production namespaces. 
For tproduction we have one F5 LB.

Procution controller:

        - --bigip-username=$(BIGIP_USERNAME)
        - --bigip-password=$(BIGIP_PASSWORD)
         - --bigip-url=https://nnn.nnn.nnn.nnn:nnn
        - --bigip-partition=KuberNetes
        - --pool-member-type=cluster
        - --namespace=tme-ns-jetty-prd
        - --namespace=tme-ns-http-prd
        - --namespace=tme-ns-nodejs-prd
        - --namespace=twistlock
        - --namespace=tme-ns-ea
        - --vs-snat-pool-name=KuberNat
        - --verify-interval=15
        - --manage-configmaps=false
        - --log-level=WARNING

So we have three F5 LBs , and for each a controller inside the 8ks clusters.

The problem we trying to solve is not something we can reproduce. As said it has ocured twice in non-production and one in production. We are now running with the controller for around 6 months.

When it happens, what I then see is that inside the F5 the some pools are red, because the members are not active anymore. Logic because new deployments were executed, meaning new pods, new IPs. And for one one reason these new IPs are not pushed to the F5 configuration. When restarting the controller is is pushing the correct information to the F5 and pools got green.

We are using cluster mode inside the controller. Meaning that we are using BGPPeering between the F5 and k8s cluster (calico network).

Walter

**Comment by wheestermans31** [comment]
Any update?

Regards
Walter

**Comment by wheestermans31** [comment]
In the meantime I have upgraded to 1.11.1 for our non-produciton environment, later this week production

**Comment by whapster** [comment]
Did the upgrade help?

**Comment by wheestermanstoyota** [comment]
No, we have upgraded to 1.11.1, and this morning project reported same issue.

I check LB configuration and some pools were red, because new deployment was executed and new pods were started, but the F5 was not updated with the new information of the started pods. Restarting the controller inside the k8s solved the issue.

We need to find the root cause of this issue asap.

Regards
Walter

**Comment by wheestermanstoyota** [comment]
We had again the issue inside our production environment.  I like to have this issue investigated with some high priority because if it happens again the issue will be escalated.

Let me know what I need to do to provide all the information when it happens again.

Walter

**Comment by ssurenr** [comment]
We will get this fixed in 1.13.

**Comment by wheestermans31** [comment]
We have some date for this?

**Comment by ssurenr** [comment]
At this time, a date is not fixed. However, it is expected in late December or early January.

**Comment by wheestermans31** [comment]
Priority of this issue should be increased because impacting our production environment!

Walter

**Comment by walter-heestermans-toyota** [comment]
Any information for possible fix data? 

Regards
Walter

**Comment by ssurenr** [comment]
Item **CONTCNTR-1395** created for Internal Tracking

**Comment by wheestermans31** [comment]
Thnaks, keep us informed about fix version/date

**Comment by ssurenr** [comment]
The symptoms looks this issue: https://github.com/F5Networks/k8s-bigip-ctlr/issues/329
Are you seeing logs related to crash in the Python process?

**Comment by Kolawole-Akinwande** [comment]
@wheestermans31 has this issue been resolved? i am experiencing it too.

**Comment by mdditt2000** [comment]
Jira issue CONTCNTR-1803, additional fixes coming in CIS 2.0 to remove this logging Overwriting existing entry for backend

**Comment by mdditt2000** [comment]
Issue was resolved in CIS 2.0. Closing Github issue out.
